Libraries/LibWeb/Painting/PaintableBox.cpp

```diff
@@ -947,7 +947,7 @@ void paint_text_fragment(DisplayListRecordingContext& context, TextPaintable con
         auto fragment_enclosing_device_rect = context.enclosing_device_rect(fragment_absolute_rect).to_type<int>();
 
         if (paintable.document().highlighted_layout_node() == &paintable.layout_node())
```

Comment:
Couldn't you just add `|| context.should_show_line_box_borders()` here at this point and remove the separate check in `PaintableWithLines::paint()`?